src/pickup.cpp

```diff
@@ -266,28 +273,24 @@ void Pickup::pick_one_up( const point &pickup_target, item &newit, vehicle *veh,
         } else if( !autopickup ) {
             // Armor can be instantly worn
             if (newit.is_armor() &&
-                query_yn(_("Put on the %s?"),
-                         newit.display_name().c_str())) {
-                if (g->u.wear_item(&newit)) {
+                query_yn(_("Put on the %s?"), newit.display_name().c_str())) {
+                if( g->u.wear_item( &newit)) {
                     picked_up = true;
                 }
             } else if (g->u.is_armed()) {
                 if (!g->u.weapon.has_flag("NO_UNWIELD")) {
-                     if( !offered_swap ) {
+                    if( !offered_swap ) {
                         offered_swap = true;
-                        if ( g->u.weapon.type->id != newit.type->id &&
-                             query_yn(_("No space for %s; wield instead? (drops %s)"),
```

Comment:
In several places you've un-wrapped long lines, making them over 100 characters long, that line wrapping was intentional, we don't want lines over 100 characters so it remains easy to edit.  Also un-wrapping them here has made the diff harder to read because there's extraneous changes mixed in with the logic changes.